easybuild/framework/easyblock.py

```diff
@@ -2453,13 +2457,26 @@ def build_and_install_one(ecdict, init_env):
                 repo_spec = spec
 
             try:
-                # upload spec to central repository
-                currentbuildstats = app.cfg['buildstats']
                 repo = init_repository(get_repository(), get_repositorypath())
+                full_ver = det_full_ec_version(app.cfg)
+                currentbuildstats = []
+
+                # grab build stats from archived easyconfig file, if it exists
+                path_in_repo = repo.easyconfig_path_for(repo_spec, app.name, full_ver)
+                if os.path.exists(path_in_repo):
+                    archived_buildstats = EasyConfigParser(path_in_repo).get_config_dict().get('buildstats')
+                    if archived_buildstats:
+                        currentbuildstats.extend(archived_buildstats)
+
+                if app.cfg['buildstats']:
+                    currentbuildstats.extend(app.cfg['buildstats'])
```

Comment:
it does, indeed

before, we were already retained the `buildstats` present in the easyconfig file being used for the installation

with this PR, we'll now also retain `buildstats` from the archived easyconfig as well

I need to look into which `buildstats` element is picked up to pick a walltime for `--job`...